### PR TITLE
Add available component builder so we can use it much easilly

### PR DESCRIPTION
**Pull request by karelhala**
To register new group you will have to add new item with name, location, title and empty array of components. When registering available component use custom decorator above some class to register it:
```Typescript
@ComponentDemo({
  name: 'tree-view-new',
  title: 'TreView',
  template: require('./../views/tree-view/basic.html'),
  controller: 'demoTreeView as vm'
})
export default class TreeViewController {
  //Demo treeView controller implementation
}
```

If you need more components registered under same controller use `AvailableCmp` multiple times as seen at [demo/controllers/dataTableController.ts](https://github.com/ManageIQ/ui-components/pull/127/files#diff-5884d947210a1d2824f3a7a10e833088)
```Typescript
@ComponentDemo({
  name: 'small',
  title: 'Small tile',
  template: require('./../views/tile-view/small.html'),
  group: 'tile-menu',
  controller: 'demoDataTable as vm'
})
@ComponentDemo({
  name: 'big',
  title: 'Big tile',
  template: require('./../views/tile-view/big.html'),
  group: 'tile-menu',
  controller: 'demoDataTable as vm'
})
@ComponentDemo({
  name: 'basic',
  title: 'Basic data table',
  template: require('./../views/data-table/basic.html'),
  group: 'data-table',
  controller: 'demoDataTable as vm'
})
export default class DataTableController {
  //Implementation of DataTablecontroller
}
```

### CODE REVIEW

**Comment by karelhala** [comment]
@miq-bot add_label enhancement
@skateman @mtho11 @himdel @Hyperkid123 what you think?

**Comment by skateman** [comment]
Well, for me it's just more unreadable :disappointed:

**Comment by karelhala** [comment]
It's just a suggestion, if you do not like we don't have to use. But @Hyperkid123 had problem with creating new component and he had some item missing so I thought someone might like this approach.

**Comment by Hyperkid123** [comment]
This seems to be much better solution for people with goofy eyes (like me).

**Comment by skateman** [comment]
Can't we ruby-ize this by dynamically loading all the html files that will have some kind of special comments at the beginning of each file? Like this:
/*
 * controller: XY controller as vm
 * url: bleh
 * section: XY components
 * name: XY1
 */

**Comment by himdel** [comment]
I like the idea, but hate that syntax... (and having to write a method for each param)

What about something simpler like..

```js
groupFactory({
  name: 'tree-view',
  title: 'Tree Components',
  location: '/tree',
  components: [
    componentFactory({
      controller: 'demoTreeView as vm',
      location: '/tree-view-new',
      name: 'tree-view-new',
      title: 'TreView',
      template: require('./../views/tree-view/basic.html'),
    }),
  ],
});
```

**Comment by karelhala** [comment]
Truth to be told the it's not easy to read either. What would you say about Ts decorators? Above example would be written like:
```Typescript
@AvailableCmp({
  name: 'tree-view-new',
  title: 'TreView',
  template: require('./../views/tree-view/basic.html'),
  controller: 'demoTreeView as vm'
})
export default class TreeViewController {
  //Demo treeView controller implementation
}
```

**Comment by karelhala** [comment]
@himdel @Hyperkid123 @mtho11 what do you say about these custom decorators?

**Comment by himdel** [comment]
Just one question .. what is AvailableCmp? :)

Wouldn't something like `ComponentDemo` make more sense?

(I mean, I get that the menu is called available components but .. that name doesn't seem to have anything to do with what this does, does it?)

**Comment by karelhala** [comment]
> Wouldn't something like ComponentDemo make more sense?

Good point. Renaming it to `ComponentDemo` decorator.

**Comment by Hyperkid123** [comment]
I like it

File: demo/services/availableComponentBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import {AvailableComponent, default as AvailableComponentsService} from './availableComponentsService';
```
Comment (skateman) [inline demo/services/availableComponentBuilder.ts]: You probably want to rename this accordingly to the new name :wink:

Comment (karelhala) [reply demo/services/availableComponentBuilder.ts]: That's ok, it's used in `componentFactory` and in decorator function.

**Comment by karelhala** [comment]
@himdel @skateman what do you think about it?

**Comment by himdel** [comment]
@karelhala LGTM, is this ready to merge? :)

But .. can you please update the description to fit the new names?

**Comment by karelhala** [comment]
@himdel updated

**Comment by himdel** [comment]
Merging, this should only affect the demo - no release concerns, right?

**Pull request merged**